processing/src/main/java/io/druid/query/filter/DimFilter.java

```diff
@@ -75,4 +77,9 @@
    * determine for this DimFilter.
    */
   RangeSet<String> getDimensionRangeSet(String dimension);
+
+  /**
+   * @return a HashSet that represents all columns' name which the DimFilter required to do filter.
+   */
+  HashSet<String> getRequiredColumns();
```

Comment:
Can we have the return type be `Set<String>` instead so that single column cases can just return `ImmutableSet.of(col)` ?

Reply:
I'm making this change to use `Set` as part of #8209.